### PR TITLE
Speed up Jet Producer for HLT

### PR DESCRIPTION
**Pull request by VinInn**
factor 2 speed up mostly by inlining area computation.
Solid 0.6% total speedup of HLT at PU22.
The JetProducer is in the need of more cleanup
(and tower revertexing is still utterly slow)
A more serious speedup could be obtained applying a cutoff to the jet energy...

No regression expected offline as the code in question runs only at HLT...

@fwyzard @gennai

### CODE REVIEW

**Comment by VinInn** [comment]
@cmsbuild , please test

File: RecoJets/JetProducers/plugins/VirtualJetProducer.cc [hunk 1]
```diff
@@ -669,87 +668,86 @@ void VirtualJetProducer::writeJets( edm::Event & iEvent, edm::EventSetup const&
...
-  auto etaFromXYZ = [](float x, float y, float z)->float { float t(z/std::sqrt(x*x+y*y)); return vdt::fast_logf(t + std::sqrt(t*t+1.f));};
-  for (auto ijet=0U;ijet<fjJets_.size();++ijet) {
-     float x = fjJets_[ijet].px();
-     float y = fjJets_[ijet].py();
-     float z = fjJets_[ijet].pz();
-     phiJ[ijet] = vdt::fast_atan2(y,x);
-     etaJ[ijet] =etaFromXYZ(x,y,z);
-   } 
+
+  auto orParam_ = 1./rParam_;
```
Comment (cvuosalo) [inline RecoJets/JetProducers/plugins/VirtualJetProducer.cc]: Can protection be added here or in the constructor to ensure that `rParam != 0`? Passing a zero value for this parameter seems like a pretty easy error for someone to make.

**Comment by cvuosalo** [comment]
Code changes are satisfactory, and Jenkins tests show no significant differences.

File: RecoJets/JetProducers/plugins/VirtualJetProducer.cc (hunk 1 above)
Comment (VinInn) [inline RecoJets/JetProducers/plugins/VirtualJetProducer.cc]: Has been like this for 15 years.
Up to the code owners to act

**Comment by cvuosalo** [comment]
@VinInn: You said: "No regression expected offline as the code in question runs only at HLT..." But I found that HLT for standard workflow 25202 does speed up, so your new code does run offline.

**Comment by cvuosalo** [comment]
+1

For #14179 0c61faea6e5e8ffe2302492c9d8ad57dc7520f50

Speeding up the jet producer for HLT.

The code changes are satisfactory, and Jenkins tests against baseline CMSSW_8_1_X_2016-04-20-1100 show no significant differences. A test of workflow 25202.0_TTbar_13 with 70 events against baseline CMSSW_8_1_0_pre3 also shows no significant differences. HLT modules that use the jet producer show reduced CPU time on the order of 50%:
```
  delta/mean delta/orJob     original                   new       module name
  ---------- ------------     --------                  ----       ------------
   -0.591829      -0.01%        10.53 ms/ev ->         5.72 ms/ev hltAK4PFJets
   -0.588334      -0.01%        10.49 ms/ev ->         5.72 ms/ev hltAK4PFJetsForTaus
   -0.521268      -0.00%         6.18 ms/ev ->         3.62 ms/ev hltAK4CaloJets
```

**Comment by davidlange6** [comment]
+1

**Pull request merged**